modules/services/window-managers/hyprland.nix

```
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.wayland.windowManager.hyprland;
+  defaultHyprlandPackage = pkgs.hyprland.override {
+    enableXWayland = cfg.xwayland.enable;
+    hidpiXWayland = cfg.xwayland.hidpi;
+    nvidiaPatches = cfg.nvidiaPatches;
+  };
+in {
+  meta.maintainers = [ lib.maintainers.fufexan ];
+
+  options.wayland.windowManager.hyprland = {
+    enable = lib.mkEnableOption (lib.mdDoc "Hyprland wayland compositor");
+
+    package = lib.mkOption {
+      type = with lib.types; nullOr package;
+      default = defaultHyprlandPackage;
+      description = lib.mdDoc ''
+        Hyprland package to use. Will override the 'xwayland' and 'nvidiaPatches'
+        options.
+      '';
+    };
+
+    plugins = lib.mkOption {
+      type = with lib.types; listOf (either package path);
+      default = [ ];
+      description = lib.mdDoc ''
+        List of Hyprland plugins to use. Can either be packages or
+        absolute plugin paths.
+      '';
+    };
+
+    systemdIntegration = lib.mkOption {
+      type = lib.types.bool;
+      default = pkgs.stdenv.isLinux;
+      description = lib.mdDoc ''
+        Whether to enable {path}`hyprland-session.target` on
+        hyprland startup. This links to `graphical-session.target`.
+        Some important environment variables will be imported to systemd
+        and dbus user environment before reaching the target, including
+        - `DISPLAY`
+        - `HYPRLAND_INSTANCE_SIGNATURE`
+        - `WAYLAND_DISPLAY`
+        - `XDG_CURRENT_DESKTOP`
+      '';
+    };
+
+    disableAutoreload = lib.mkOption {
```

Best to avoid confusion with double negation and such. I would suggest

```suggestion
    enableAutoreload = lib.mkOption {
```

with the default being `true`. That said, is there any occasion where you would not want automatic reload?

---

I added it because of [this issue](https://github.com/hyprwm/Hyprland/issues/1255), however I think it's safe to remove it.

---

Removed.